File: 0009-palindrome-number/0009-palindrome-number.c
bool isPalindrome(int x) {
    int rem,check,temp;
    temp=x;
    check=0;
    if(x<0 || x!=0 && x%10 ==0 ) return false;
    while(x>check){
            check = check*10 + x%10;
            x/=10;
        }
    return (x==check || x==check/10);

}

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int temp=x;
        int y=0;
        while(x>0){
            int rem=x%10;
            y=(y*10)+rem;
            x=x/10;
        }
        return temp==y;
    }
}